stdext/result.mli

```diff
@@ -27,4 +27,6 @@ module List : sig
 
   val fold_left :
     'a list -> f:('acc -> 'a -> ('acc, 'c) t) -> init:'acc -> ('acc, 'c) t
+
+  val exists : 'a list -> f:('a -> (bool, 'err) t) -> (bool, 'err) t
```

Comment:
Could you add a doc string to document the behavior on empty lists? It matches `List.exists` and `Base.List.exists` but I think it would be useful to just have it spelled out directly.